librad/src/git/types.rs

```diff
@@ -115,14 +115,14 @@ impl<R> SymbolicRef<R> {
 }
 
 #[derive(Clone)]
-pub struct Refspec<R> {
-    pub(crate) remote: SomeReference<R>,
-    pub(crate) local: SomeReference<R>,
+pub struct Refspec<RemoteR, LocalR> {
```

Comment:
Arrghhh I think I want these to be existential as well...

Reply:
But I think that might just make things more of a pain in the ass because I'd have 2 `Either`s so twice amount of case handling.

Reply:
Maybe having a `trait AsRefspec` would work and that way instead of using `Vec<Refspec<R, L>` we would have `Vec<Box<dyn AsRefspec>>`. To give an idea of what I'm thinking I'd like something like:
```rust
pub struct Remote {
  url: String, // possibly make Url
  name: String,
  fetch_specs: Vec<Box<dyn AsRefspec>>,
  push_specs: Vec<Box<dyn AsRefspec>>,
}
```